src/hooks/tradingView/useChartLines.ts

```diff
@@ -327,17 +327,25 @@ export const useChartLines = ({
     ]
   );
 
+  useEffect(
+    () => {
+      if (initialWidget && !isChartReady) {
+        // Clear lines when chart switches to not ready after initialization (i.e. when orderbookCandles is toggled)
```

Comment:
seems like we are vulnerable to other race conditions and things because of how this is setup and awkwardly syncing between react and trading view. I'm not sure how to reason about this in a way that guarantees we're always in sync correctly.

Reply:
yeahhh I don't love it :( the best solution (which avoids race conditions altogether) would be to be able to draw multiple series on the tradingView chart simultaneously (one with the orderbook values, and one with trading values) and then toggle hiding / showing each one based on the toggle val. I couldn't get it to cleanly work though and couldn't find any good TradingView documentation around how to do so 🫠 but happy to file a follow up to spend more time investigating here if this proves too icky!!